ballerina/os.bal

```diff
@@ -58,3 +58,54 @@ public isolated function getUserHome() returns string = @java:Method {
     name: "getUserHome",
     'class: "io.ballerina.stdlib.os.nativeimpl.GetUserHome"
 } external;
+
+# Sets the value of the environment variable named by the key. 
+# ```ballerina
+# os:Error? err = os:setEnv("BALCONFIGFILE", "/path/to/Config.toml");
+# ```
+#
+# + key - Key of the environment variable
+# + value - Value of the environment variable
+# + return - error if setting the environment variable fails, () otherwise
+public isolated function setEnv(string key, string value) returns Error? {
+    if key == "" {
+        return error Error("key cannot be an empty string");
+    } else if key == "==" {
+        return error Error("key cannot be == sign");
+    } else if key == "0x00" {
+        return error Error("key cannot be initial hexadecimal zero character (0x00)");
```

Comment:
Don't we need to avoid signs as well? any particular reason avoid these two

Reply:
Checked the behavior in golang. These 2 characters are allowed. I think we only need to avoid empty strings.

Reply:
Will add validation to check if the key has non-alpha-numeric characters.